src/protections/ProtectionManager.ts

```diff
@@ -146,9 +135,21 @@ export class ProtectionManager {
     get protections(): Readonly<Map<string /* protection name */, Protection>> {
         return this._protections;
     }
+    private readonly protection_classes: Protection[] = [
```

Comment:
NIT: Let's stick to the naming convention of `protectionClasses` please, sorry if that's been butchered in some other place.